packages/ngtools/webpack/src/transformers/find_image_domains_spec.ts

```diff
@@ -0,0 +1,180 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { tags } from '@angular-devkit/core';
+import * as ts from 'typescript';
+import { findImageDomains } from './find_image_domains';
+import { createTypescriptContext, transformTypescript } from './spec_helpers';
+
+function findDomains(
+  input: string,
+  importHelpers = true,
+  module: ts.ModuleKind = ts.ModuleKind.ES2020,
+) {
+  const { program, compilerHost } = createTypescriptContext(input, undefined, undefined, {
+    importHelpers,
+    module,
+  });
+  const domains = new Set<string>();
+  const getTypeChecker = () => program.getTypeChecker();
+  const transformer = findImageDomains(getTypeChecker, domains);
+
+  transformTypescript(input, [transformer], program, compilerHost);
+
+  return domains;
+}
+
+function inputTemplate(provider: string) {
+  /* eslint-disable max-len */
+  return tags.stripIndent`
+    export class AppModule {
```

Comment:
I'd like to propose that we add a test for standalone bootstrap pattern as well (though the `providers` array structure would be the same), so that we make sure it works for standalone cases. An input can look something like this:

```
bootstrapApplication(AppComponent, {
  providers: [ /* the format here is similar to what you have in other tests*/ ]
});
```